src/ir/context.rs

```diff
@@ -772,4 +773,108 @@ impl<'ctx> BindgenContext<'ctx> {
 
         self.current_module = previous_id;
     }
+
+    /// Iterate over all (explicitly or transitively) whitelisted items.
+    ///
+    /// If no items are explicitly whitelisted, then all items are considered
+    /// whitelisted.
+    pub fn whitelisted_items<'me>(&'me self) -> WhitelistedItemsIter<'me, 'ctx> {
+        assert!(self.in_codegen_phase());
+        assert!(self.current_module == self.root_module);
+
+        let roots = self.items()
+            .filter(|&(_, item)| {
+                // If nothing is explicitly whitelisted, then everything is fair
+                // game.
+                if self.options().whitelisted_types.is_empty() &&
+                   self.options().whitelisted_functions.is_empty() &&
+                   self.options().whitelisted_vars.is_empty() {
+                    return true;
+                }
+
+                let name = item.canonical_name(self);
+                match *item.kind() {
+                    ItemKind::Module(..) => false,
+                    ItemKind::Function(_) => self.options().whitelisted_functions.matches(&name),
+                    ItemKind::Var(_) => self.options().whitelisted_vars.matches(&name),
+                    ItemKind::Type(ref ty) => {
+                        if self.options().whitelisted_types.matches(&name) {
+                            return true;
+                        }
+
+                        // Unnamed top-level enums are special and we whitelist
+                        // them via the `whitelisted_vars` filter, since they're
+                        // effectively top-level constants, and there's no way
+                        // for them to be referenced consistently.
+                        if let TypeKind::Enum(ref enum_) = *ty.kind() {
+                            if ty.name().is_none() &&
+                               enum_.variants().iter().any(|variant| {
+                                self.options().whitelisted_vars.matches(&variant.name())
+                            }) {
+                                return true;
+                            }
+                        }
+
+                        false
+                    }
+                }
+            })
+            .map(|(&id, _)| id);
+
+        let seen: ItemSet = roots.collect();
+        let to_iterate = seen.iter().cloned().collect();
+
+        WhitelistedItemsIter {
+            ctx: self,
+            seen: seen,
+            to_iterate: to_iterate,
+        }
+    }
+}
+
+/// An iterator over whitelisted items.
+///
+/// See `BindgenContext::whitelisted_items` for more information.
+pub struct WhitelistedItemsIter<'ctx, 'gen>
+    where 'gen: 'ctx
+{
+    ctx: &'ctx BindgenContext<'gen>,
+
+    // The set of whitelisted items we have seen. If you think of traversing
```

Comment:
It isn't `pub` so I'm not sure that it matters -- do you feel strongly about it?